Packs/ipinfo/Integrations/ipinfo_v2/ipinfo_v2.py

```diff
@@ -0,0 +1,197 @@
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+
+import requests
+import traceback
+from typing import Dict, Any
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+
+class Client(BaseClient):
+    def __init__(self, api_key: str, base_url: str, verify_certificate: bool, proxy: bool,
+                 reliability: DBotScoreReliability):
+        """
+        Client to use in the IPinfo integration. Uses BaseClient
+        """
+        super().__init__(base_url=base_url, proxy=proxy, verify=verify_certificate)
+        self.api_key = api_key
+        self.reliability = reliability
+
+    def _url_suffix(self, ip: str) -> str:
+        api_key_in_url = f'?token={self.api_key}' if self.api_key else ''
+        return f'{ip}/json{api_key_in_url}'
+
+    def ipinfo_ip(self, ip: str) -> Dict[str, Any]:
+        return self._http_request(method='GET', url_suffix=self._url_suffix(ip=ip))
+
+
+def test_module(client: Client) -> str:
+    """Tests IPinfo by sending a query on 8.8.8.8"""
+    client.ipinfo_ip('8.8.8.8')
+    return 'ok'  # on any failure, an exception is raised
+
+
+#
+
+def ipinfo_ip_command(client: Client, ip: str) -> List[CommandResults]:
+    response = client.ipinfo_ip(ip)
+    return parse_results(ip, response, client.reliability)
+
+
+def parse_results(ip: str, raw_result: Dict[str, Any], reliability: DBotScoreReliability) -> List[CommandResults]:
+    command_results = []
+
+    # default values
+    asn = as_owner = None
+    feed_related_indicators = []
+
+    if raw_result:
+        hostname = raw_result.get('hostname')
+        feed_related_indicators.append(
+            Common.FeedRelatedIndicators(hostname,
+                                         FeedIndicatorType.URL if urlRegex.find(hostname)
+                                         else FeedIndicatorType.Domain))
+        if 'asn' in raw_result:
+            asn = demisto.get(raw_result, 'asn.asn')
+            as_owner = demisto.get(raw_result, 'asn.name')
+            as_domain = demisto.get(raw_result, 'asn.domain')
+
+            if as_domain:
+                feed_related_indicators.append(Common.FeedRelatedIndicators(as_domain, FeedIndicatorType.Domain))
+
+        elif 'org' in raw_result:
```

Comment:
`asn` and `org` are mutually exclusive?

Reply:
v1 looks so, what do you think?

https://github.com/demisto/content/blob/e1c26879d309aa84c5aa27c30a0ff3bc4f7905d8/Packs/ipinfo/Integrations/integration-Ipinfo.yml#L65-L79

Reply:
@dorschw please consult with the API, and test this.

Reply:
ok, got it: the free plan returns ` org,` while the "basic" (simplest paid plan) returns both `org` and,`asn` the latter having `asn.name` with an _identical value_ to the one under `org` (at least for the couple examples I tried) 

so the logic behind v1 seems to have been "if you have more data, show it, otherwise default to only having `org`